diff --git a/src/logging.cc b/src/logging.cc
index 4028ccc..2151ef4 100644
--- a/src/logging.cc
+++ b/src/logging.cc
@@ -2348,7 +2348,7 @@ const vector<string>& GetLoggingDirectories() {
       if (GetWindowsDirectoryA(tmp, MAX_PATH))
         logging_directories_list->push_back(tmp);
       logging_directories_list->push_back(".\\");
-#else
++#elif !defined(__ANDROID__)
       logging_directories_list->push_back("./");
 #endif
     }


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 846b444..7de30e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -615,6 +615,8 @@ add_library (glogbase OBJECT
   ${GLOG_SRCS}
 )
 
+target_compile_options(glogbase PRIVATE "-ffile-prefix-map=${CMAKE_CURRENT_SOURCE_DIR}=.")
+
 add_library (glog
   $<TARGET_OBJECTS:glogbase>
 )
